File: programmers/level2/pro42588.py
# 프로그래머스 탑

def solution(heights):
    answer = [0 for i in range(len(heights))]
    for i in range(len(heights)-1, 0, -1): # 마지막 부터 해서
        for j in range(i-1, -1, -1): # 그앞에껄로 차근차근 해서 
            if(heights[i] < heights[j]): # 더크면 answer에 그 값을 저장.
                answer[i] = j+1
                break
    return answer

File: javascript/pro42588.js
// 프로그래머스 탑 

function solution(heights) {
    const answer = [];
    for (let i = heights.length-1; i >= 0; i--) {
        let check = false;
        for (let j = i-1; j >= 0; j--) {
            if(heights[i] < heights[j]) {
                answer.unshift(j+1);
                check = true;
                break;
            }
        }
        if (!check) {
            answer.unshift(0);
        }
    }
    return answer;
}

function solution1(heights) {
    return heights.map((v, i) => {
        while (i) {
            i--;
            if (heights[i] > v) {
                return i + 1;
            }
        }
        return 0;
    })
}